src/main/java/de/tum/in/www1/artemis/service/ProgrammingSubmissionService.java

```diff
@@ -676,21 +681,36 @@ public ProgrammingSubmission lockAndGetProgrammingSubmission(Long submissionId)
      * @return a locked programming submission that needs an assessment
      */
     public ProgrammingSubmission lockAndGetProgrammingSubmissionWithoutResult(ProgrammingExercise exercise, long correctionRound) {
-        ProgrammingSubmission programmingSubmission = getRandomProgrammingSubmissionEligibleForNewAssessment(exercise, exercise.hasExerciseGroup(), correctionRound)
+        ProgrammingSubmission programmingSubmission = getRandomProgrammingSubmissionEligibleForNewAssessment(exercise, exercise.isExamExercise(), correctionRound)
                 .orElseThrow(() -> new EntityNotFoundException("Programming submission for exercise " + exercise.getId() + " could not be found"));
-        Result newManualResult = lockSubmission(programmingSubmission);
+        Result newManualResult = lockSubmission(programmingSubmission, correctionRound);
         return (ProgrammingSubmission) newManualResult.getSubmission();
     }
 
+    // TODO SE: explain in what context this method is even called
+
+    /**
+     * Locks the programmingSubmission submission. If the submission only has automatic results, and no manual,
+     * create a new manual result. In the second correction round add a second manual result to the submission
+     *
+     * @param submission the submission to lock
+     * @param correctionRound
+     * @return
+     */
     @Override
-    protected Result lockSubmission(Submission submission) {
-        Result automaticResult = submission.getLatestResult();
-        List<Feedback> automaticFeedbacks = automaticResult.getFeedbacks().stream().map(Feedback::copyFeedback).collect(Collectors.toList());
-        // Create a new result (manual result) and a new submission for it and set assessor and type to manual
-        ProgrammingSubmission newSubmission = createSubmissionWithLastCommitHashForParticipation((ProgrammingExerciseStudentParticipation) submission.getParticipation(),
-                SubmissionType.MANUAL);
-
-        Result newResult = saveNewEmptyResult(newSubmission);
+    protected Result lockSubmission(Submission submission, Long correctionRound) {
+
+        Result existingResult;
+        if (correctionRound == 0 && submission.getLatestResult().getAssessmentType().equals(AssessmentType.AUTOMATIC)) {
+            existingResult = submission.getLatestResult();
+        }
+        else {
+            existingResult = submission.getResultByCorrectionRoundIgnoreAutomatic(correctionRound - 1);
```

Comment:
If I understand this correctly: When `correctionRound == 0` and the latest result is not automatic, we will try to get results from correction round `-1`?